packages/sdk/src/TokenboundClient.ts

```diff
@@ -418,22 +431,45 @@ class TokenboundClient {
     data: `0x${string}`
     // operation?: CallOperation // The type of operation to perform ( CALL: 0, DELEGATECALL: 1, CREATE: 2, CREATE2: 3)
   }> {
-    const { account, to, value, data } = params
+    const { account, to, value, data, chainId = this.chainId } = params
     const operation = CALL_OPERATIONS.CALL
 
     if (!this.supportsV3) {
       // const { operation, ...rest } = params
       return await this.prepareExecuteCall(params)
     }
 
+    let executionArgs = [to, value, data, operation]
+
+    // Handle cross-chain call encoding
+    if (this.chainId !== chainId) {
+      const {
+        to: crossChainTo,
+        value: crossChainValue,
+        data: crossChainData,
+      } = await encodeCrossChainCall({
+        publicClient: this.publicClient,
+        account,
+        to,
+        value,
+        data: data as `0x${string}`,
```

Comment:
Looks like we should tighten up typing on `ExecuteCallParams` in `params.ts` like so:

```
  data: `0x${string}`
```

so we don't necessitate explicit casting elsewhere.